public/app/features/dashboard/services/DashboardLoaderSrv.ts

```diff
@@ -77,13 +82,16 @@ export class DashboardLoaderSrv {
           };
         });
     } else if (uid) {
-      const cachedDashboard = stateManager.getFromCache(uid);
-      if (cachedDashboard) {
-        return Promise.resolve(cachedDashboard);
+      // Don't load dashboard from cache if we're passing any query parameters
+      if (Object.keys(queryParams ?? {}).length === 0) {
+        const cachedDashboard = stateManager.getFromCache(uid);
+        if (cachedDashboard) {
+          return Promise.resolve(cachedDashboard);
+        }
```

Comment:
Technically for the same scope we should return from cache as well, as this should produce a determnistic dashboard. So I think we should use a combination of uid and scopes to form a cache key.